lte/gateway/python/magma/pipelined/tests/test_ipv6_solicitation_5g.py

```diff
@@ -0,0 +1,185 @@
+    PipelinedController,
+    TestSetup,
+)
+from magma.pipelined.tests.app.table_isolation import (
+    RyuDirectTableIsolator,
+    RyuForwardFlowArgsBuilder,
+)
+from magma.pipelined.tests.pipelined_test_util import (
+    SnapshotVerifier,
+    create_service_manager,
+    start_ryu_app_thread,
+    stop_ryu_app_thread,
+    wait_after_send,
+)
+from scapy.arch import get_if_hwaddr
+from scapy.layers.inet6 import (
+    ICMPv6ND_NS,
+    ICMPv6ND_RS,
+    ICMPv6NDOptSrcLLAddr,
+    IPv6,
+)
+from scapy.layers.l2 import Ether
+
+
+class IPV6RouterSolicitation5GTableTest(unittest.TestCase):
+    BRIDGE = 'testing_br'
+    IFACE = 'testing_br'
+    MAC_DEST = "5e:cc:cc:b1:49:4b"
+    BRIDGE_IP = '192.168.128.1'
+    UE_BLOCK = '192.168.128.0/24'
+    UE_MAC = '5e:cc:cc:b1:49:4b'
+    UE_IP = '192.168.128.22'
+    OTHER_MAC = '0a:00:27:00:00:02'
+    OTHER_IP = '1.2.3.4'
+
+    @classmethod
+    def setUpClass(cls, *_):
+        """
+        Starts the thread which launches ryu apps
+
+        Create a testing bridge, add a port, setup the port interfaces. Then
+        launch the ryu apps for testing pipelined. Gets the references
+        to apps launched by using futures.
+        """
+        super(IPV6RouterSolicitation5GTableTest, cls).setUpClass()
+        warnings.simplefilter('ignore')
+        cls.service_manager = create_service_manager(
+            [],
+            ['ipv6_solicitation'],
+        )
+        cls._tbl_num = cls.service_manager.get_table_num(IPV6SolicitationController.APP_NAME)
+
+        ipv6_controller_reference = Future()
+        testing_controller_reference = Future()
+        test_setup = TestSetup(
+            apps=[
+                PipelinedController.IPV6RouterSolicitation,
+                PipelinedController.Testing,
+                PipelinedController.StartupFlows,
+            ],
+            references={
+                PipelinedController.IPV6RouterSolicitation:
+                    ipv6_controller_reference,
+                PipelinedController.Testing:
+                    testing_controller_reference,
+                PipelinedController.StartupFlows:
+                    Future(),
+            },
+            config={
+                'setup_type': 'LTE',
+                'allow_unknown_arps': False,
+                'bridge_name': cls.BRIDGE,
+                'bridge_ip_address': cls.BRIDGE_IP,
+                'ovs_gtp_port_number': 32768,
+                'virtual_interface': cls.BRIDGE,
+                'local_ue_eth_addr': True,
+                'quota_check_ip': '1.2.3.4',
+                'ipv6_router_addr': 'd88d:aba4:472f:fc95:7e7d:8457:5301:ebce',
+                'clean_restart': True,
+                'virtual_mac': 'd6:34:bc:81:5d:40',
+                'enable_nat': True,
+                'classifier_controller_id': 5,
+                'enable5g_features': True,
+            },
+            mconfig=PipelineD(
+                ue_ip_block=cls.UE_BLOCK,
```

Comment:
does not seem to be needed for the test

Reply:
Yes, but need to defined one mconfig parameter otherwise getting below error:
TypeError: __init__() missing 1 required positional argument: 'mconfig'

Reply:
maybe @nstng was saying that `mconfig` is unused. Does `PipelineD(..)` have side effects?

Reply:
The `ue_ip_block` argument can be removed - not the complete `mconfig` argument.

I will close conversation anyway and move further discussion to the general test simplification discussion.

Reply:
Yes, we can remove the ue_ip_block argument but it's not a functionality issue.
Ok, I removed it for now.